### PR TITLE
members modal: generalise to be reusable

### PR DESCRIPTION
**Pull request by anikachurilova**
* disable users and groups that are already a part of community or have already been invited
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2593

Screenshot 2024-03-15 at 16 57 29

Screenshot 2024-03-15 at 16 57 50

inviteee

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/GroupTabPane.js [hunk 1]
```diff
@@ -80,6 +109,10 @@ export class GroupTabPane extends Component {
                 handleChange={this.addMemberToSelected}
                 searchType="group"
                 placeholder={i18next.t("Search for groups")}
+                existingEntities={existingEntitiesIds}
+                existingEntitiesDescription={i18next.t(
+                  "This group is a member or has already been invited to this community."
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/GroupTabPane.js:114]: Groups are not invited but simply added:
```suggestion
                  "Already a member"
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/InvitationsMembersModal.js [hunk 2]
```diff
@@ -14,35 +14,71 @@ import { withState } from "react-searchkit";
...
+import { RichEditor, withCancel, http } from "react-invenio-forms";
 
 export class InvitationsMembersModal extends Component {
   constructor(props) {
     super(props);
     this.state = {
       open: false,
       activeIndex: 0, // by default members is the active pane
+      message: undefined,
+      existing: undefined,
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/InvitationsMembersModal.js:27]: Apply same changes above

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/InvitationsMembersModal.js [hunk 3]
```diff
@@ -63,12 +99,36 @@ export class InvitationsMembersModal extends Component {
...
-          <MembersWithRoleSelection
+          <SearchWithRoleSelection
             key="members-users"
             roleOptions={userRoles}
             modalClose={this.handleCloseModal}
             action={api.createInvite}
             onSuccessCallback={this.onMemberSuccess}
+            searchBarTitle={<label>{i18next.t("Member")}</label>}
+            searchBarTooltip={i18next.t(
+              'Users must set profile visibility to "Public" in order to be invited to a community.'
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/InvitationsMembersModal.js:110]: I think we should be a little bit more vague, something like:
```suggestion
              'Search for users to invite (only users with a public profile can be invited).'
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/MemberSearchBar.js [hunk 4]
```diff
@@ -37,44 +37,58 @@ export class MembersSearchBar extends Component {
   };
 
   serializeUsersForDropdown = (users) => {
+    const { existingEntitiesDescription } = this.props;
```
Comment (ntarocco) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/invitationsModal/MemberSearchBar.js:40]: Check the linter msg above

**Review by ntarocco — COMMENTED** [review]
Nice work! Have a look to the linter warnings.

It feels weird to have the SearchWithRolesSelection reusable common cmp here in `invenio-communities`, but I guess there is no better option at the moment.

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/SearchWithRoleSelection.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      messageComponent,
+      existingEntities,
+      existingEntitiesDescription,
+    } = this.props;
+    const { selected, loading, error } = this.state;
+    const selectedCount = Object.keys(selected).length;
+
+    return (
+      <>
+        <div className="rel-pl-2 rel-pr-2 rel-pb-2 rel-pt-2">
```
Comment (kpsherva) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/SearchWithRoleSelection.js:78]: if I remember correctly this should do the job:
```suggestion
        <div className="default-padding">
```

**Review by kpsherva — APPROVED** [review]

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/SearchWithRoleSelection.js (hunk 5 above)
Comment (anikachurilova) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/SearchWithRoleSelection.js]: You are right! Updated

**Pull request merged**